src/integer_mod_q/polynomial_ring_zq/properties.rs

```diff
@@ -0,0 +1,152 @@
+    ///
+    /// Returns `true` if the polynomial is irreducible and `false` otherwise.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::PolynomialRingZq;
+    /// use std::str::FromStr;
+    ///
+    /// let poly_irr = PolynomialRingZq::from_str("2  1 1 / 3  1 2 3 mod 17").unwrap();
+    /// // returns true, since X + 1 is irreducible
+    /// assert!(poly_irr.is_irreducible());
+    /// ```
+    pub fn is_irreducible(&self) -> bool {
+        let poly_over_zq = PolyOverZq::from((&self.poly, self.modulus.get_q()));
+
+        // Since after reduction with the polynomial modulus, the check of
+        // irreducibility is the same as without the polynomial modulus
+        poly_over_zq.is_irreducible()
+    }
+
+    /// Checks if a [`PolynomialRingZq`] is the constant polynomial with coefficient `1`.
+    ///
+    /// Returns `true` if there is only one coefficient, which is `1`.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::PolynomialRingZq;
+    /// use std::str::FromStr;
+    ///
+    /// let value = PolynomialRingZq::from_str("1  1 / 3  1 0 1 mod 4").unwrap();
+    /// assert!(value.is_one());
+    /// ```
+    pub fn is_one(&self) -> bool {
+        self.poly.is_one()
+    }
+
+    /// Checks if every entry of a [`PolynomialRingZq`] is `0`.
+    ///
+    /// Returns `true` if [`PolynomialRingZq`] has no coefficients.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer_mod_q::PolynomialRingZq;
+    /// use std::str::FromStr;
+    ///
+    /// let value = PolynomialRingZq::from_str("0 / 2  1 1 mod 7").unwrap();
+    /// assert!(value.is_zero());
+    /// ```
+    pub fn is_zero(&self) -> bool {
+        self.poly.is_zero()
+    }
+}
+
+#[cfg(test)]
+mod test_is_irreducible {
+    use crate::integer_mod_q::PolynomialRingZq;
+    use std::str::FromStr;
+
+    /// Ensure that a irreducible [`PolynomialRingZq`] returns `true`.
+    #[test]
+    fn poly_is_irreducible() {
+        // 9X^2 + 12X + 10 is irreducible over 17
+        let poly_irr = PolynomialRingZq::from_str("3  10 12 9 / 4  1 10 12 9 mod 17").unwrap();
+        print!("{}", poly_irr);
+        assert!(poly_irr.is_irreducible());
+    }
+
+    /// Ensure that a reducible [`PolynomialRingZq`] returns `false`.
+    #[test]
+    fn poly_is_reducible() {
+        let poly_irr = PolynomialRingZq::from_str("3  1 2 1 / 4  1 10 12 9 mod 17").unwrap();
+        assert!(!poly_irr.is_irreducible());
+    }
+}
+
+#[cfg(test)]
+mod test_is_one {
+    use super::PolynomialRingZq;
+    use std::str::FromStr;
+
+    /// Ensure that is_one returns `true` for the one polynomial.
+    #[test]
+    fn one_detection() {
+        let one = PolynomialRingZq::from_str("1  1 / 4  1 10 12 9 mod 7").unwrap();
+        let one_2 = PolynomialRingZq::from_str("2  1 14 / 4  1 10 12 9 mod 7").unwrap();
+
+        assert!(one.is_one());
+        assert!(one_2.is_one());
+    }
```

Comment:
Please add a test that ensures reduction by the polynomial, e.g. `3 11 4 10 / 5 2 5 mod 20` should work (I think).